internal/gojs/state.go

```diff
@@ -195,6 +195,9 @@ func (j *values) increment(v interface{}) uint32 {
 }
 
 func (j *values) decrement(id uint32) {
+	if id < nextID {
```

Comment:
> it seems like decrement for values below nextID should be a
nop or decrement should not be called at all.

I would love to put a comment with the permalink for your research here
```suggestion
	// Decrement  for values below nextID should be a
nop.
	// See https://github.com/golang/go/....
	if id < nextID {
```

Reply:
This wasn't based on research into the golang repo's `wasm_exec.js`, but purely
based on wazero's gojs implementation. I couldn't actually find any references
in the upstream implementation to nextID or the reason the `value` structs's
get/increment/decrement in `state.go` uses it to offset IDs before lookup.

Looking at what increments the refcount, it only happens in `storeRef`.
However, it does not increment for values that are *jsVal:
https://github.com/tetratelabs/wazero/blob/d5664ed5d1f5af2836c29ff731a1f13192bf255d/internal/gojs/state.go#L133-L134
which jsfs is:
https://github.com/tetratelabs/wazero/blob/d5664ed5d1f5af2836c29ff731a1f13192bf255d/internal/gojs/fs.go#L21.
based on the switch statement in
https://github.com/tetratelabs/wazero/blob/d5664ed5d1f5af2836c29ff731a1f13192bf255d/internal/gojs/state.go#L55,
all ids < nextID are special cased and not stored in values.

I think the values struct was only meant to to handle ids >= nextID, and that
finalize was not expected to be called for ids < nextID. I did not find the
reason why it is called on a low id, but since `wasm_exec.js` doesn't differentiate
among the ids on the finalizer it seems reasonable that it could. If wazero
special cases the load/store for these ids, it probably also should the
finalize.
 
Happy to commenting something along these lines or moving the id check to the
finalize call instead, but I don't have an external source to cite.

Reply:
ok ic, thank you for clarifying. I think I defer this to @codefromthecrypt as this is beyond my knowledge on gojs.

Reply:
```suggestion
	// Special IDs are not refcounted.
	if id < nextID {
```

Reply:
I added a comment which while imperfect is better than nothing. The overall thing is that we have more "special IDs" than go has, to avoid constant thrash on things that are commonly requested, and won't change for the same module instance. the guest is unaware of these special IDs so they will try to call finalize etc for them. special casing is the correct thing to do here!